cucumber-android/src/main/java/cucumber/runtime/android/CucumberExecutor.java

```diff
@@ -123,13 +124,20 @@ public CucumberExecutor(final Arguments arguments, final Instrumentation instrum
             feature.sendTestSourceRead(bus);
         }
         this.pickleEvents = FeatureCompiler.compile(features, filters);
-        instrumentationReporter.setNumberOfTests(getNumberOfConcreteScenarios());
+    }
+
+    /**
+     * Only for internal usage: exposes the list of future tests
+     */
+    List<PickleEvent> getPickleEvents() {
+        return pickleEvents;
     }
 
     /**
      * Runs the cucumber scenarios with the specified arguments.
      */
     public void execute() {
+        instrumentationReporter.setNumberOfTests(getNumberOfConcreteScenarios());
```

Comment:
we need to defer a moment of reporting of total tests to right before the execution, because then filtering of tests can be applied:
https://github.com/cucumber/cucumber-android/pull/14/commits/fcd70d3f333c276f087b1b94a1e5b579614dfbe0#diff-9d77d6e39a9caffbbd7c7854efbb2fe0R67